openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2897,132 +2904,233 @@ class GBSAParameterHandler(ParameterHandler):
+                                           cutoff=amber_cutoff,
+                                           kappa=0,
+                                           )
+            # WARNING: If using a CustomAmberGBForce, the functional form is affected by whether
+            # the cutoff kwarg is None *during initialization*. So, if you initialize it with a
+            # non-None value, and then try to change it to None, you'll basically break things.
+
+        # Set the GBSAForce to have the same cutoff as NonbondedForce
+        #gbsa_force.setCutoffDistance(nonbonded_force.getCutoffDistance())
+        if amber_cutoff is not None:
+            gbsa_force.setCutoffDistance(amber_cutoff)
+
+        #gbsa_force.setNonbondedMethod(nonbonded_force.getNonbondedMethod())
+
+        if nonbonded_force.usesPeriodicBoundaryConditions():
+            # WARNING: The lines below aren't equivalent. The NonbondedForce and
+            # CustomGBForce NonbondedMethod enums have different meanings.
+            # More details:
+            # http://docs.openmm.org/latest/api-python/generated/simtk.openmm.openmm.NonbondedForce.html
+            # http://docs.openmm.org/latest/api-python/generated/simtk.openmm.openmm.GBSAOBCForce.html
+            # http://docs.openmm.org/latest/api-python/generated/simtk.openmm.openmm.CustomGBForce.html
+
+            #gbsa_force.setNonbondedMethod(simtk.openmm.NonbondedForce.CutoffPeriodic)
+            gbsa_force.setNonbondedMethod(simtk.openmm.CustomGBForce.CutoffPeriodic)
+        else:
+            #gbsa_force.setNonbondedMethod(simtk.openmm.NonbondedForce.NoCutoff)
+            gbsa_force.setNonbondedMethod(simtk.openmm.CustomGBForce.NoCutoff)
+
+        # Add all GBSA terms to the system. Note that this will have been done above
+        if self.gb_model == 'OBC2':
+            gbsa_force.setSolventDielectric(self.solvent_dielectric)
+            gbsa_force.setSoluteDielectric(self.solute_dielectric)
+            if self.sa_model is None:
+                gbsa_force.setSurfaceAreaEnergy(0)
+            else:
+                gbsa_force.setSurfaceAreaEnergy(self.surface_area_penalty)
+
+
+        # Iterate over all defined GBSA types, allowing later matches to override earlier ones.
+        atom_matches = self.find_matches(topology)
+
+        # Create all particles.
+
+        # !!! WARNING: CustomAmberGBForceBase expects different per-particle parameters
+        # depending on whether you use addParticle or setParticleParameters. In
+        # setParticleParameters, we have to apply the offset and scale BEFORE setting
+        # parameters, whereas in addParticle, the offset is applied automatically, and the particle
+        # parameters are not set until an auxillary finalize() method is called. !!!
+
+        # To keep it simple, we DO NOT pre-populate the particles in the GBSA force here.
+        # We call addParticle further below instead.
+        #for topology_particle in topology.topology_particles:
```

Comment:
is this meant to be commented out?

Reply:
Yes. This was a huge footgun that I fired many times when I'd try to come back to this after a break. I've left copious warnings for future inhabitants. I might propose leaving a larger, more lasting warning as well: https://en.wikipedia.org/wiki/Long-time_nuclear_waste_warning_messages

Reply:
Yes, it's a really confusing API.  Sorry about that!  I didn't design it. :/